src/sources/internal_logs.rs

```diff
@@ -25,6 +29,17 @@ inventory::submit! {
 
 impl_generate_config_from_default!(InternalLogsConfig);
 
+impl InternalLogsConfig {
+    /// Return an internal metrics config with enterprise reporting defaults.
+    pub fn enterprise(version: impl Into<String>, configuration_key: impl Into<String>) -> Self {
```

Comment:
Rust-noob question, is there a difference/preference between
```
    pub fn enterprise(version: impl Into<String>, configuration_key: impl Into<String>) -> Self {
```
and 
```
    pub fn enterprise<T: Into<String>>(version: T, configuration_key: T) -> Self {
```

Reply:
In the first case, `version` and `configuration_key` can not be the same type but in the second case they have to be the same type.
For using `impl Into<String>` instead of the `<T: Into<String>>`, I don't really know what that implies. I'd be curious to know about that.

For this piece of code in particular, I just reproduced what @leebenson did for the `InternalMetricsConfig`.